code/game/objects/structures/mineral_doors.dm

```diff
@@ -83,26 +83,30 @@
 		icon_state = "[base_icon_state][smoothing_flags ? "-[smoothing_junction]" : ""]"
 
 
-/obj/structure/mineral_door/attackby(obj/item/W, mob/living/user)
+/obj/structure/mineral_door/attackby(obj/item/attacking_item, mob/living/user)
 	. = ..()
 	if(.)
 		return
 	if(QDELETED(src))
 		return
 
+	calculate_attackby(attacking_item, user)
+
+///Handles damage calculation + plasmacutter interaction from attackby
+/obj/structure/mineral_door/proc/calculate_attackby(obj/item/attacking_item, mob/living/user)
```

Comment:
this isn't very clear. This isn't a calculation proc, its doing actual damage, specifically if an item is doing burn damage.